.github/workflows/pre-commit.yaml

```
@@ -0,0 +1,126 @@
+    types: [opened, synchronize, reopened]
+
+jobs:
+  precommit:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
```

actionlint (1.7.9)

the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)